### ISSUE TITLE
Prediction scores of ranked lists are all equal

### ISSUE THREAD

**Issue by rickyyue** [tensorflow/ranking#126]
I'm following the case of "Using sparse features and embeddings in TF-Ranking" to build a LTR model with meta(dense) features and sparse text features in both context_feature_spec and example_feature_spec. For text features I have text length ranging from 100 ~ 1000 words. 
I use an ordinal label(0 - 7) with 7 as the most relevant. Below is what I found:

1. the loss fluctuates between -0.5 and -0.7 when training
2. the predicted score on both training and test set for each ranked list are identical as shown below:
array([-1.3086238e-07, -1.3086238e-07, -1.3086236e-07, -1.3086235e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07, -1.3086236e-07, -1.3086236e-07,
       -1.3086236e-07, -1.3086236e-07], dtype=float32)
3. For training: 'metric/ndcg@1': 0.28329352, 'metric/ndcg@10': 0.3729706,

I have roughly 450K rows(from 9K groups) for training and 45K(900 groups) for test. 
I tried tuning the learning rate from 1-e7 to 1 and _embedding_dim from 20 to 300. I used a vocabulary size of 30000 to embed context and example text features. Besides I tried batch size of 32 and 64. But I sended up with getting identical predictions scores for all the tries. 

Can anyone suggest any solution to improve my model performance? 
Has anyone tried Bert or any other pre-training token embedding for sparse features? Any improvement?

Thank you.

Ricky

**Comment by xuanhuiwang** [comment]
For the identical predictions, my guess is that all your features end up with the default values: [] for sparse features and 0 for dense features. It may be worth checking your feature_spec.

Also, we recently released a script version for tfrecord data: https://github.com/tensorflow/ranking/blob/master/tensorflow_ranking/examples/tf_ranking_tfrecord.py. You may try this out as well.

**Comment by xuanhuiwang** [comment]
Close now. Feel free to reopen.
